alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/clustering/ClusteringMetricsBeanPostProcessor.java

```diff
@@ -0,0 +1,72 @@
+package eu.xenit.alfred.telemetry.binder.clustering;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.NoSuchBeanDefinitionException;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.config.ConstructorArgumentValues;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.beans.factory.support.GenericBeanDefinition;
+
+public class ClusteringMetricsBeanPostProcessor implements BeanDefinitionRegistryPostProcessor {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClusteringMetricsBeanPostProcessor.class);
+
+    public static final String CLUSTER_SERVICE = "clusterService";
```

Comment:
We most likely want to use the `ClusterService` bean. 

In Alfresco there are public services and their "low level" equivalents. 
Public services are the "low level" services wrapped with typically some interceptors, e.g. for security. 

In general, the "low level" services has the same bean name, but starting with a small character. 

E.g. ClusterService (public) vs clusterService (low level), NodeService vs nodeService, ...

Reply:
In this case I dont think so since there is a bean registered under that name that is coming from a sun proxy
Bean name: ClusterService
Bean class: class com.sun.proxy.$Proxy172

Reply:
Yes, but the proxy does implement the `ClusterService` interface:

```
    <bean id="ClusterService" class="org.springframework.aop.framework.ProxyFactoryBean">
        <property name="proxyInterfaces">
            <list>
                <value>org.alfresco.enterprise.repo.cluster.core.ClusterService</value>
            </list>
        </property>
        <property name="target">
            <ref bean="clusterService"/>
        </property>
        <property name="interceptorNames">
            <list>
                <idref bean="clusterServiceTxInterceptor"/>
            </list>
        </property>
    </bean>
```
(`cluster-context.xml`)

When using that bean, the proxy will "act like it's a ClusterService". Invocations will proceed to the low level service / the proxied service.

Reply:
What might be a problem though, is that there's also a `ClusterService` bean available in ACE:

```
    <bean id="ClusterService" class="java.lang.String"/>
```
(`minimal-context.xml`)

So no `NoSuchBeanDefinitionException` will be thrown, though it might not be the best idea to use the public service here.